dask_bigquery/core.py

```diff
@@ -174,3 +200,108 @@ def make_create_read_session_request(row_filter="", max_stream_count=0):
         return new_dd_object(graph, output_name, meta, divisions)
+
+
+def to_gbq(
+    df,
+    *,
+    project_id: str,
+    dataset_id: str,
+    table_id: str,
+    gs_bucket: str = "dask-bigquery-tmp",
+    credentials: Credentials = None,
+    delete_bucket: bool = False,
+    parquet_kwargs: dict = None,
+    load_api_kwargs: dict = None,
+):
+    """Write dask dataframe as table using BigQuery Load API.
+    Writes Parquet to GCS for intermediary storage.
+
+    Parameters
+    ----------
+    project_id: str
+      Name of the BigQuery project id.
+    dataset_id: str
+      BigQuery dataset within project
+    table_id: str
+      BigQuery table within dataset
+    gs_bucket: str
+      Google Cloud Storage bucket name, for intermediary parquet storage. If the bucket doesn't
+      exist, it will be created. The account you're using will need Google Cloud Storage
+      permissions (storage.objects.create, storage.buckets.create). If a persistent bucket is used,
+      it might be a good idea to configure it with retention, to make sure the temporary data gets
+      cleaned periodically, even if the job is failed to finish and cleanup.
+      Default: dask-bigquery-tmp.
+    credentials : google.auth.credentials.Credentials, optional
+      Credentials for accessing Google APIs. Use this parameter to override
+      default credentials.
+    delete_bucket:
+      Delete bucket in GCS after loading intermediary data to Big Query.
+      Default: False.
+    parquet_kwargs: dict
+      Additional kwargs to pass to dataframe.write_parquet, such as schema, partition_on or
+      write_index. For writing parquet, pyarrow is required.
+      Default: {"write_index": False}
+    load_api_kwargs: dict
+      Additional kwargs to pass when creating bigquery.LoadJobConfig, such as schema,
+      time_partitioning, clustering_fields, etc.
+      Default: {"autodetect": True}
+
+    Returns
+    -------
+        LoadJobResult
+    """
+    if project_id is None and hasattr(credentials, "project_id"):
+        # service account credentials have a project associated with them
+        project_id = credentials.project_id
+    if not project_id:
+        raise ValueError("Required: project_id")
+    if not table_id:
+        raise ValueError("Required: table_id")
+    if not dataset_id:
+        raise ValueError("Required: dataset_id")
+
+    parquet_kwargs = parquet_kwargs or {}
+    if not parquet_kwargs:
+        parquet_kwargs["write_index"] = False
+
+    load_api_kwargs = load_api_kwargs or {}
+    if not load_api_kwargs:
+        load_api_kwargs["autodetect"] = True
+
+    fs = gcs_fs(project_id, credentials=credentials)
+    if not fs.exists(gs_bucket):
+        fs.mkdir(gs_bucket)
+
+    token = tokenize(df)
+    object_prefix = f"{dt.datetime.utcnow():%Y-%m-%dT%H-%M-%S}_{token}"
+
+    path = f"gs://{gs_bucket}/{object_prefix}"
+    df.to_parquet(
+        path=path, engine="pyarrow", write_metadata_file=False, **parquet_kwargs
+    )
```

Comment:
this should be part of the try/except block. otherwise the cleanup won't work if the dask run fails